cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -209,12 +213,12 @@ private Scheduler getSchedulerForSchedule(Id.Program program, SchedulableProgram
                                                String scheduleName) throws NotFoundException {
     ApplicationSpecification appSpec = getStore().getApplication(program.getApplication());
     if (appSpec == null) {
-      throw new NotFoundException("application", program.getApplicationId());
+      throw new ApplicationNotFoundException(program.getApplicationId());
     }
 
     Map<String, ScheduleSpecification> schedules = appSpec.getSchedules();
     if (schedules == null || !schedules.containsKey(scheduleName)) {
-      throw new NotFoundException("schedule", scheduleName);
+      throw new ScheduleNotFoundException(scheduleName);
```

Comment:
In this whole class (and in general, in this new schedule code), exceptions are not handled consistently. Sometimes we throw something meaningful (like a ScheduleNotFoundExn), sometimes, we propagate any exception as a RuntimeException. Sometimes we log, sometimes we don't. You would not think that this code was written by one person.